app/models/MovementListSearchOptions.scala

```diff
@@ -47,41 +109,173 @@ case class MovementListSearchOptions(searchKey: Option[MovementSearchSelectOptio
 
 object MovementListSearchOptions {
 
+  private[models] def localDateToString(ld: LocalDate): String =
+    s"${f"${ld.getDayOfMonth}%02d"}/${f"${ld.getMonthValue}%02d"}/${f"${ld.getYear}%04d"}" // pad day, month and year with leading zeros as needed
+
+  private[models] def stringToLocalDate(s: String): LocalDate =
+    LocalDate.parse(s, DateTimeFormatter.ofPattern("dd/MM/yyyy"))
+
   val DEFAULT_INDEX: Int = 1
   val DEFAULT_MAX_ROWS: Int = 10
 
+  object CHOOSE_PRODUCT_CODE extends SelectOptionModel {
+    override val code: String = "chooseProductCode"
+    override val displayName: String = "viewAllMovements.filters.exciseProductCode.chooseProductCode"
+  }
+
+  object CHOOSE_COUNTRY extends SelectOptionModel {
+    override val code: String = "chooseCountry"
+    override val displayName: String = "viewAllMovements.filters.countryOfOrigin.chooseCountry"
+  }
+
   implicit def queryStringBinder(implicit intBinder: QueryStringBindable[Int],
-                                 stringBinder: QueryStringBindable[String]): QueryStringBindable[MovementListSearchOptions] =
+                                 stringBinder: QueryStringBindable[String]
+                                ): QueryStringBindable[MovementListSearchOptions] =
     new QueryStringBindable[MovementListSearchOptions] {
 
-      override def bind(key: String, params: Map[String, Seq[String]]): Option[Either[String, MovementListSearchOptions]] =
+      override def bind(key: String, params: Map[String, Seq[String]]): Option[Either[String, MovementListSearchOptions]] = {
         Some(for {
           sortOrder <- stringBinder.bind("sortBy", params).getOrElse(Right(ArcAscending.code))
           index <- intBinder.bind("index", params).getOrElse(Right(DEFAULT_INDEX))
           searchKey <- stringBinder.bind("searchKey", params).map(_.map(Some(_))).getOrElse(Right(None))
           searchValue <- stringBinder.bind("searchValue", params).map(_.map(Some(_))).getOrElse(Right(None))
+          traderRole <- stringBinder.bind("traderRole", params).map(_.map(Some(_))).getOrElse(Right(None))
```

Comment:
Could you write a small helper method for the calls on each of these lines after `stringBinder.bind`?